query/src/test/scala/filodb/query/exec/rangefn/InstantFunctionSpec.scala

```diff
@@ -335,7 +335,12 @@ class InstantFunctionSpec extends RawDataWindowingSpec with ScalaFutures {
             else if (val1.isNaN) val2.isNaN shouldEqual true
             else val1 shouldEqual val2 +- 0.0001
         }
+        // The zip method appears to short circuit and doesn't always fully drain all the
+        // iterators. This can leave locks stuck. Calling size has the side effect of fully
+        // draining the iterator, which will in turn ensure that locks are released.
+        ex.size
+        res.size
```

Comment:
Can we close cursor instead of draining it ? Better yet, we should replace the NoCloseCursor in the test with custom close call.

Reply:
I have no idea where the cursor is coming from. The tests that fail aren't creating cursors directly as far as I can tell. I'll have to take heap dump and reverse engineer the code.